### PR TITLE
Implement GetZoneByProviderID and GetZoneByNodeName for GCE

### PR DESCRIPTION
**Pull request by FengyunPan**
This is part of #50926
cc @wlan0

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by wlan0** [comment]
/assign @cheftako 

The code change looks good

File: pkg/cloudprovider/providers/gce/gce_zones.go [hunk 1]
```diff
@@ -43,14 +42,35 @@ func (gce *GCECloud) GetZone() (cloudprovider.Zone, error) {
...
 func (gce *GCECloud) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByProviderID not implemented")
+	_, zone, _, err := splitProviderID(providerID)
+	if err != nil {
+		return cloudprovider.Zone{}, err
+	}
+
+	gceZone := cloudprovider.Zone{
+		FailureDomain: zone,
+		Region:        gce.region,
```
Comment (cheftako) [inline pkg/cloudprovider/providers/gce/gce_zones.go]: Please use GetGCERegion(zone)

File: pkg/cloudprovider/providers/gce/gce_zones.go [hunk 2]
```diff
@@ -43,14 +42,35 @@ func (gce *GCECloud) GetZone() (cloudprovider.Zone, error) {
...
-	return cloudprovider.Zone{}, errors.New("GetZoneByNodeName not imeplemented")
+	instanceName := mapNodeNameToInstanceName(nodeName)
+	instance, err := gce.getInstanceByName(instanceName)
+	if err != nil {
+		return cloudprovider.Zone{}, err
+	}
+
+	gceZone := cloudprovider.Zone{
+		FailureDomain: instance.Zone,
+		Region:        gce.region,
```
Comment (cheftako) [inline pkg/cloudprovider/providers/gce/gce_zones.go]: GetGCERegion(instance.zone)

**Review by cheftako — COMMENTED** [review]
Minor tweaks.

**Comment by FengyunPan** [comment]
@cheftako I find that you create a new PR to implement GetZoneByProviderID and GetZoneByNodeName for GCE, #51561, so I close my PR.

**Pull request closed without merging**